src/TropicalGeometry/variety_supertype.jl

```diff
@@ -0,0 +1,590 @@
+function pm_object(v::TropicalVarietySupertype)
+  return v.polymakeTV
+end
+
+
+###
+# 1. Basic constructions
+# ----------------------
+###
+
+@doc Markdown.doc"""
+    intersect(T1, T2)
+
+Intersect two tropical varieties.
+
+# Examples
+```jldoctest
+julia> RR = tropical_numbers(min)
+Tropical ring (min)
+
+julia> S,(x,y) = RR["x","y"]
+(Multivariate Polynomial Ring in x, y over Tropical ring (min), AbstractAlgebra.Generic.MPoly{Oscar.TropicalNumbersElem{typeof(min)}}[x, y])
+
+julia> f1 = x+y+1
+x + y + (1)
+
+julia> f2 = x^2+y^2+T(-6)
+x^2 + y^2 + (-6)
+
+julia> hyp1 = TropicalHypersurface(f1)
+A min tropical hypersurface embedded in 2-dimensional Euclidian space
+
+julia> hyp2 = TropicalHypersurface(f2)
+A min tropical hypersurface embedded in 2-dimensional Euclidian space
+
+julia> tv12 = intersect(hyp1, hyp2)
+A min tropical variety of dimension 1 embedded in 2-dimensional Euclidian space
+```
+"""
+function intersect(T1::TropicalVarietySupertype{M, EMB}, T2::TropicalVarietySupertype{M, EMB}) where {M, EMB}
+    return TropicalVariety{M, EMB}(intersect(T1.polyhedralComplex,T2.polyhedralComplex))
+end
+
+
+@doc Markdown.doc"""
+    intersect_stably(T1, T2)
+
+# Examples
+```jldoctest
+```
+"""
+function stably_intersect(T1::TropicalVarietySupertype{M, EMB}, T2::TropicalVarietySupertype{M, EMB}) where {M, EMB}
+    return TropicalVariety{M, EMB}(intersect_stably(T1.polyhedralComplex,T2.polyhedralComplex))
+end
+export intersect_stably
+
+
+
+###
+# 2. Basic properties
+# -------------------
+###
+
+@doc Markdown.doc"""
+    ambient_dim(T::TropicalVariety{M, EMB})
+    ambient_dim(T::TropicalCurve{M, EMB})
+    ambient_dim(T::TropicalHypersurface{M, EMB})
+    ambient_dim(T::TropicalLinearSpace{M, EMB})
+
+Returns the ambient dimension of `T` if it is embedded. Returns an error otherwise.
+
+# Examples
+A tropical hypersurface in RR^n is of ambient dimension n
+```jldoctest
+julia> RR = tropical_numbers(min);
+
+julia> S,(x,y) = RR["x","y"];
+
+julia> f = x+y+1;
+
+julia> tropicalLine = TropicalHypersurface(f);
+
+julia> ambient_dim(tropicalLine)
+# todo: add examples for varieties, curves and linear spaces
+```
+"""
+function ambient_dim(T::TropicalVarietySupertype{M,EMB}) where {M,EMB}
+    if !EMB
+        error("ambient_dim: tropical variety not embedded")
+    end
+
+    return ambient_dim(T.polyhedralComplex)
+end
+
+
+
+@doc Markdown.doc"""
+    codim(T::TropicalVariety{M, EMB})
+    codim(T::TropicalCurve{M, EMB})
+    codim(T::TropicalHypersurface{M, EMB})
+    codim(T::TropicalLinearSpace{M, EMB})
+
+Returns the codimension of `T`.
+
+# Examples
+A tropical hypersurface in RR^n is always of dimension n-1
+```jldoctest
+julia> RR = tropical_numbers(min);
+
+julia> S,(x,y) = RR["x","y"];
+
+julia> f = x+y+1;
+
+julia> tropicalLine = TropicalHypersurface(f);
+
+julia> codim(tropicalLine)
+# todo: add examples for varieties, curves and linear spaces
+```
+"""
+function codim(T::TropicalVarietySupertype{M,EMB}) where {M,EMB}
```

Comment:
In places where you don't need to access the type parameters, you can simply omit them. I've just picked this one example, but of course it applies in many other places.
```suggestion
function codim(T::TropicalVarietySupertype)
```

Reply:
Thanks, I will apply this on the new PR.